src/arena.cc

```diff
@@ -0,0 +1,35 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "arena.h"
+
+#include <algorithm>
+
+char* Arena::AllocSlowPath(size_t num_bytes)
+{
+  size_t to_allocate = std::max(next_size_, num_bytes);
+
+  Block new_block;
+  new_block.mem.reset(new char[to_allocate]);
```

Comment:
nit: Used an aligned new here since there is technically no guarantee that this will happen (yes, I know most allocators would use size_t as a minimum). For now the arena is only used to store character sequences so this doesn't matter, but this could become problematic if it is used later to allocate other things.

Reply:
Good point. In theory you should be able to use a dedicated type with `alignas()` specifier, and use this in a `new AlignedType[(to_allocate + sizeof(AlignedType) - 1) & ~sizeof(AlignedType)])`, but this is getting ugly, and frankly there is no need for this feature here. What do you think about dropping the alignment requirement entirely, and just call this `StringPieceArena` for clarity?

Reply:
The arena is taken out now, so we'll deal with it in a separate PR if we get there.